libexec/rtld-elf/rtld_c18n.c

```diff
@@ -770,11 +770,11 @@ resize_table(int exp)
 }
 
 void
-tramp_hook_impl(void *, int, void *, const Obj_Entry *, const Elf_Sym *,
+tramp_hook_impl(uintptr_t *, int, void *, const Obj_Entry *, const Elf_Sym *,
     void *);
 
 void
-tramp_hook_impl(void *rcsp, int event, void *target, const Obj_Entry *obj,
+tramp_hook_impl(uintptr_t *rcsp, int event, void *target, const Obj_Entry *obj,
```

Comment:
Is rcsp actually logically a pointer to an array of uintptr_t, or is this just being done for convenience because you don't want to cast in the function? The latter isn't valid justification, the former is, but then you should change the other instances in the file that talk about rcsp too. As it stands the file as a whole is inconsistent.